kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopEventManager.java

```diff
@@ -288,54 +289,74 @@ void process(BiConsumer<String, Long> registerEventLatency,
         @Override
         public void process(BiConsumer<String, Long> registerEventLatency,
                             long startProcessTime) {
-            if (!coordinator.isActive()) {
-                return;
-            }
-
             try {
                 List<String> topicsDeletions = metadataStore.getChildren(getDeleteTopicsPath()).get();
 
-                HashSet<String> topicsFullNameDeletionsSets = Sets.newHashSet();
-                HashSet<KopTopic> kopTopicsSet = Sets.newHashSet();
-                topicsDeletions.forEach(topic -> {
-                    KopTopic kopTopic = new KopTopic(topic);
-                    kopTopicsSet.add(kopTopic);
-                    topicsFullNameDeletionsSets.add(kopTopic.getFullName());
-                });
-
                 if (log.isDebugEnabled()) {
                     log.debug("Delete topics listener fired for topics {} to be deleted", topicsDeletions);
                 }
 
-                Iterable<GroupMetadata> groupMetadataIterable = coordinator.getGroupManager().currentGroups();
-                HashSet<TopicPartition> topicPartitionsToBeDeletions = Sets.newHashSet();
-
-                groupMetadataIterable.forEach(groupMetadata -> {
-                    topicPartitionsToBeDeletions.addAll(
-                            groupMetadata.collectPartitionsWithTopics(topicsFullNameDeletionsSets));
-                });
-
-                Set<String> deletedTopics = Sets.newHashSet();
-                if (!topicPartitionsToBeDeletions.isEmpty()) {
-                    coordinator.handleDeletedPartitions(topicPartitionsToBeDeletions);
-                    Set<String> collectDeleteTopics = topicPartitionsToBeDeletions
-                            .stream()
-                            .map(TopicPartition::topic)
-                            .collect(Collectors.toSet());
-
-                    deletedTopics = kopTopicsSet.stream().filter(
-                            kopTopic -> collectDeleteTopics.contains(kopTopic.getFullName())
-                    ).map(KopTopic::getOriginalName).collect(Collectors.toSet());
-
-                    deletedTopics.forEach(deletedTopic -> {
-                        metadataStore.delete(
-                                getDeleteTopicsPath() + "/" + deletedTopic, Optional.of((long) -1));
-                    });
-                }
+                // Localize groupCoordinatorsByTenant to avoid multi-thread conflicts
+                final Map<String, GroupCoordinator> currentCoordinators = groupCoordinatorsByTenant;
```

Comment:
This is not a copy operation

Reply:
You are right. I will use a copy operation to create a new Map. Thanks very much.